### PR TITLE
[mlir][tensor] Improve tensor.pack simplication pattern. 

### PR DESCRIPTION
**Pull request by hanhanW**
A tensor.pack op can be rewritten to a tensor.expand_shape op if the packing only happens on inner most dimension.

This also formats the lit checks better.

### CODE REVIEW

**Comment by hanhanW** [comment]
It depends on https://github.com/llvm/llvm-project/pull/76606, please take a look at https://github.com/llvm/llvm-project/pull/76606/commits/9cf7bb7d05f95601e746c431be6ea32a8bc5c37f or wait until the parent PR is landed.

**Review by chelini — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**
